compiler/src/dmd/dsymbolsem.d

```diff
@@ -1558,6 +1581,12 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
             foreach (s; (*pd.decl)[])
             {
+                if (pd.ident == Id.printf || pd.ident == Id.scanf)
+                {
+                    s.setPragmaPrintf(pd.ident == Id.printf);
+                    continue;
+                }
+
                 s.dsymbolSemantic(sc2);
```

Comment:
@dkorpel how important is it to run `setPragmaPrintf` before `dsymbolSemantic`?

Reply:
When setPragmaPrintf is done after dsymbolSenantic, the varargs signature isn't checked

Reply:
OK, see #14717 then.